lib/diagnostic.js

```diff
@@ -40,7 +44,8 @@ const hasNone = function (array, callback) {
 // you may use any feature of JavaScript
 // to solve this problem
 const last = function (array, callback) {
-
+  const newArray = array.filter(callback)
+  return newArray[newArray.length - 1]
```

Comment:
Careful, what you are doing here is calling the function `newArray` inside your function called `newArray`. Also, you want to make sure that the callback function is actually returning `true`.